### PR TITLE
feat(server): Account for serializer's temporary buffer size

### PR DESCRIPTION
**Pull request by chakaz**

### CODE REVIEW

File: src/server/snapshot.cc [hunk 1]
```diff
@@ -364,6 +364,14 @@ size_t SliceSnapshot::GetTotalChannelCapacity() const {
   return dest_->GetSize();
 }
 
+size_t SliceSnapshot::GetTempBuffersSize() const {
```
Comment (romange) [inline src/server/snapshot.cc:367]: I think in this context "temp buffer" is an implementation detail. Maybe call it "MemoryUsed" instead?

Comment (romange) [reply src/server/snapshot.cc:367]: Now I see there are three functions in SliceSnapshot and it's not easy to understand how they are different in semantics.
nit: I would prefer that `SerializerBase::GetTempBuffersSize` name will be different from that of SliceSnapshot

Comment (chakaz) [reply src/server/snapshot.cc:367]: Removed the term `Total` and added a comment :pray:

**Comment by chakaz** [comment]
> you have not pushed imho

:facepalm:

**Review by romange — APPROVED** [review]

**Pull request merged**